torch/distributed/tensor/_utils.py

```diff
@@ -30,19 +32,29 @@ class ExplicitRedistributionContext:
     during forward and perform a manual redistribution during backwards.
     """
 
-    _explicit_redistribute_mode = False
+    _local = threading.local()
+    _local._stack = []
+
+    def __init__(self, enable: bool = True, strict: bool = False):
+        self._enable = enable
+        self._strict = strict
 
     @classmethod
-    def is_active(cls) -> bool:
-        return cls._explicit_redistribute_mode
+    def is_redistribute_allowed(cls, src_spec: DTensorSpec, dst_spec: DTensorSpec):
+        if len(cls._local._stack):
+            instance = cls._local._stack[-1]
+            if instance._enable:
+                if instance._strict:
+                    return False
+                return redistribute_cost(src_spec, dst_spec) == 0
```

Comment:
maybe 
```suggestion
                return redistribute_cost(src_spec, dst_spec) <= 0
```
given I had https://github.com/pytorch/pytorch/blob/main/torch/distributed/tensor/_ops/_tensor_ops.py#L240